data_juicer/ops/mapper/mllm_mapper.py

```diff
@@ -0,0 +1,113 @@
+from data_juicer.ops.op_fusion import LOADED_IMAGES
+from data_juicer.utils.mm_utils import load_image
+from data_juicer.utils.model_utils import get_model, prepare_model
+
+OP_NAME = 'mllm_mapper'
+torch.set_num_threads(1)
+
+
+@LOADED_IMAGES.register_module(OP_NAME)
+@OPERATORS.register_module(OP_NAME)
+class MllmMapper(Mapper):
+    """Mapper to optimize instruction.
+    Recommended model list: [
+        liuhaotia/llava-v1.6-vicuna-7b
+    ]
+    """
+    _accelerator = 'cuda'
+
+    def __init__(self,
+                 hf_model: str = 'liuhaotia/llava-v1.6-vicuna-7b',
+                 max_new_tokens=256,
+                 temperature=0.2,
+                 top_p=None,
+                 num_beams=1,
+                 *args,
+                 **kwargs):
+        """
+        Initialization method.
+        :param hf_model: hugginface model id.
+        :param max_new_tokens: the maximum number of new tokens
+            generated by the model.
+        :param temperature: used to control the randomness of \
+            generated text. The higher the temperature, the more \
+                random and creative the generated text will be.
+        :param top_p: randomly select the next word from the group \
+            of words whose cumulative probability reaches p.
+        :param num_beams: the larger the beam search size, the higher \
+            the quality of the generated text.
+        :param args: extra args
+        :param kwargs: extra args
+        """
+        super().__init__(*args, num_proc=1, **kwargs)
+
+        self.hf_model = hf_model
+        self.model_key = prepare_model(model_type='huggingface',
+                                       pretrained_model_name_or_path=hf_model)
+        self.max_new_tokens = max_new_tokens
+        self.temperature = temperature
+        self.top_p = top_p
+        self.num_beams = num_beams
+
+    def process(self, sample=None, rank=None):
+
+        # there is no image in this sample
+        if self.image_key not in sample or not sample[self.image_key]:
+            return sample
+
+        # load images
+        loaded_image_keys = sample[self.image_key]
+        images = {}
+        for loaded_image_key in loaded_image_keys:
+            if loaded_image_key not in images:
+                # avoid loading the same images
+                image = load_image(loaded_image_key)
+                images[loaded_image_key] = image
+
+        if torch.cuda.is_available():
+            model, processor = get_model(self.model_key,
+                                         rank=rank,
+                                         use_cuda=True)
+        else:
+            model, processor = get_model(self.model_key, rank=rank)
+
+        conversation = [
+            {
+                'role':
+                'user',
+                'content': [
+                    {
+                        'type': 'text',
+                        'text': sample[self.text_key]
```

Comment:
I’m a bit confused—should 'text' here refer to the original caption or a task instruction?

Reply:
It refers to a task instruction.